spec/ic.did

```diff
@@ -43,6 +62,15 @@ service ic : {
   delete_canister : (record {canister_id : canister_id}) -> ();
   deposit_cycles : (record {canister_id : canister_id}) -> ();
   raw_rand : () -> (blob);
+  http_request : (record {
+    url : text;
+    method : variant { get };
+    headers: vec http_header;
+    body : opt blob;
+    transform : opt variant {
+      function: func (http_response) -> (http_response) query
+    };
+  }) -> (variant { Ok : http_response; Err: opt http_request_error });
```

Comment:
If there is a transformation function anyways, there is little reason to return an `http_response`. Assume you use this to scrape some data off a webpage. Surely, the `transform` function will parse the result, find the relevant data (say, the temperature in Palo Alto) and will want to return _just_ that. This will also make the message much smaller.

With the current interface it would have to artificially wrap it in a `http_response`.

So an alternative is to leave the resulting type abstract. Unfortunately, [Candid does not (yet) have generics](https://github.com/dfinity/candid/issues/245), so `blob` might be a better choice:
```suggestion
      function: func (http_response) -> (blob) query
    };
  }) -> (variant { Ok : blob; Err: opt http_request_error });
```

This would make the transform _required_, but in practice, I woudln’t be surprised if any real use will need a `transform` anyways. Most APIs, for example, return headers indicating throttling information. So while toy do get more complicated.

Of course, `http_response` contains a `blob` in the body, and this could be (ab?)used instead.

Reply:
Given that in the specifications is mentioned that the canister can modify headers too,  I will leave `http_response` as a return type here.